src/providers/alibaba.ts

```diff
@@ -0,0 +1,77 @@
+import type { ApiProvider, ProviderOptions } from '../types';
+import type { EnvOverrides } from '../types/env';
+import { OpenAiChatCompletionProvider, OpenAiEmbeddingProvider } from './openai';
+
+const KNOWN_MODELS = new Set([
+  // Flagship models
+  'qwen-max',
+  'qwen-max-latest',
+  'qwen-max-2025-01-25',
+  'qwen-plus',
+  'qwen-plus-latest',
+  'qwen-plus-2025-01-25',
+  'qwen-turbo',
+  'qwen-turbo-latest',
+  'qwen-turbo-2024-11-01',
+  // Visual models
+  'qwen-vl-max',
+  'qwen-vl-plus',
+  'qwen2.5-vl-72b-instruct',
+  'qwen2.5-vl-7b-instruct',
+  'qwen2.5-vl-3b-instruct',
+  // Qwen 2.5 models
+  'qwen2.5-7b-instruct-1m',
+  'qwen2.5-14b-instruct-1m',
+  'qwen2.5-72b-instruct',
+  'qwen2.5-32b-instruct',
+  'qwen2.5-14b-instruct',
+  'qwen2.5-7b-instruct',
+  // Qwen 2 models
+  'qwen2-72b-instruct',
+  'qwen2-57b-a14b-instruct',
+  'qwen2-7b-instruct',
+  // Qwen 1.5 models
+  'qwen1.5-110b-chat',
+  'qwen1.5-72b-chat',
+  'qwen1.5-32b-chat',
+  'qwen1.5-14b-chat',
+  'qwen1.5-7b-chat',
+  // Embedding models
+  'text-embedding-v3',
+]);
+
+const API_BASE_URL = 'https://dashscope-intl.aliyuncs.com/compatible-mode/v1';
+
+export function createAlibabaProvider(
```

Comment:
Not a fan of this `createXXXProvider` pattern.  It means there's an extra step when importing the providers to use them (e.g. in cloud), and it's not better in any way compared to making a class.  

Can we just implement the shared interface instead?